### PR TITLE
Fix "Others" styling

### LINKED ISSUE #7651: [Fill component] Show the 'Others' category at the bottom of the category list.

### LINKED ISSUE #8287: Category maps with 10 or more options fill all points as others.

### PR DESCRIPTION
**Pull request by javierarce**
This PR fixes an issue where creating a ramp using the fill component with more than 10 colors, produced an incorrect CartoCSS.

Before the change, styling a layer with more than 10 categories produced this CartoCSS: 
```
#layer {
  marker-width: 7;
  marker-fill: ramp([one], (#555, #5B3F95, #1D6996, #129C63, #73AF48, #EDAD08, #E17C05, #C94034, #BA0040, #8E1966, #6F3072), (Others, "one", "two", "three", "four", "five", "six", "seven", "eight", "nine", "ten"));
  marker-fill-opacity: 1;
  marker-allow-overlap: true;
  marker-line-width: 1;
  marker-line-color: #FFF;
  marker-line-opacity: 1;
}
```
That rendered this map:

screen shot 2016-09-28 at 14 42 34

After the change, we'll have:

```
#layer {
  marker-width: 7;
  marker-fill: ramp([one], (#5B3F95, #1D6996, #129C63, #73AF48, #EDAD08, #E17C05, #C94034, #BA0040, #8E1966, #6F3072, #555), ( "one", "two", "three", "four", "five", "six", "seven", "eight", "nine", "ten"));
  marker-fill-opacity: 1;
  marker-allow-overlap: true;
  marker-line-width: 1;
  marker-line-color: #FFF;
  marker-line-opacity: 1;
}
```

And the following map: 

screen shot 2016-09-28 at 14 43 27

**Updated**: The last screen capture is updated per commit https://github.com/CartoDB/cartodb/pull/9938/commits/77c106ffa2c7fb7d4566670b21c33a3cfb5f1bf7 (which puts the 'Others' category at the end of the list)

Closes #8287
Closes #7651

### CODE REVIEW

**Comment by javierarce** [comment]
CR: @xavijam

**Comment by javierarce** [comment]
I decided to change my initial PR code because I found out issue #7651, which asks us to put the "Others" category at the end of the list. This allows to simplify the code and removes the need to touch the `style-converter.js` file.

**Comment by xavijam** [comment]
Ok @javierarce, take into account one thing, we get guess Others category checking that in the style-converter (https://github.com/CartoDB/cartodb/blob/master/lib/assets/javascripts/cartodb3/editor/style/style-converter.js#L4). It is likely we need to move that "MAX_VALUES" to the new view.

**Comment by xavijam** [comment]
BTW, spotted the problem in production, let's try to prioritize it please ❤️ .

**Comment by javierarce** [comment]
Added a missing spec and ```input-color-categories-list-view.js``` uses the ```MAX_VALUES``` defined in ```input-color-categories.js```

File: lib/assets/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-color-picker/input-color-picker-header.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ module.exports = CoreView.extend({
     this.$el.append(template({
       ramp: this.model.get('ramp'),
       index: this.model.get('index'),
-      label: this._getColor().title
+      label: this._getColor().title || _t('form-components.editors.fill.input-categories.others')
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-color-picker/input-color-picker-header.js]: :+1:

**Comment by javierarce** [comment]
Ok, fixed the torque query generation and the related spec.

File: lib/assets/javascripts/cartodb3/editor/style/style-converter.js [hunk 2]
```diff
@@ -473,9 +475,7 @@ function animation (style, mapContext) {
     return v.replace(/\n/g, '\\n').replace(/\"/g, '\\"').replace();
   }
 
-  // Others is specified at the beginning of the domain, so if there are more than MAX_VALUES
-  // the loop will start from the next pos, 1.
-  for (var i = hasOthers ? 1 : 0, l = color.domain.length; i < l; i++) {
+  for (var i = 0, l = color.domain.length; i < l; i++) {
     var categoryName = color.domain[i];
     var categoryPos = i + 1;
     var value;
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/style/style-converter.js]: Shouldn't we change the ELSE condition in the query to take the last value (11?) for the others?

File: lib/assets/test/spec/cartodb3/editor/style/style-converter-fixtures.spec.js [hunk 3]
```diff
@@ -720,7 +710,7 @@ module.exports = [
...
-        sql: 'select *, (CASE WHEN "columnA" = 2 THEN 2 WHEN "columnA" = 3 THEN 3 WHEN "columnA" = 4 THEN 4 WHEN "columnA" = 5 THEN 5 WHEN "columnA" = 6 THEN 6 WHEN "columnA" = 7 THEN 7 WHEN "columnA" = 8 THEN 8 WHEN "columnA" = 9 THEN 9 WHEN "columnA" = 10 THEN 10 WHEN "columnA" = 11 THEN 11  ELSE 1  END) as value FROM (<%= sql %>) __wrapped',
+        sql: 'select *, (CASE WHEN "columnA" = 1 THEN 1 WHEN "columnA" = 2 THEN 2 WHEN "columnA" = 3 THEN 3 WHEN "columnA" = 4 THEN 4 WHEN "columnA" = 5 THEN 5 WHEN "columnA" = 6 THEN 6 WHEN "columnA" = 7 THEN 7 WHEN "columnA" = 8 THEN 8 WHEN "columnA" = 9 THEN 9 WHEN "columnA" = 10 THEN 10 WHEN "columnA" = 11 THEN 11  ELSE 1  END) as value FROM (<%= sql %>) __wrapped',
```
Comment (xavijam) [inline lib/assets/test/spec/cartodb3/editor/style/style-converter-fixtures.spec.js]: ELSE 11 END as we have pointed ^^^ above?

File: lib/assets/test/spec/cartodb3/editor/style/style-converter-fixtures.spec.js [hunk 4]
```diff
@@ -20,12 +20,6 @@ module.exports = [
         cartocss: '#layer {\nmarker-fill: #000;\nmarker-fill-opacity: 0.4;\nmarker-allow-overlap: true;\nmarker-comp-op: darken;\n}',
```
Comment (xavijam) [inline lib/assets/test/spec/cartodb3/editor/style/style-converter-fixtures.spec.js]: I'd add a new test checking Others with categories (no torque, just a simple one).

File: lib/assets/test/spec/cartodb3/editor/style/style-converter-fixtures.spec.js (hunk 3 above)
Comment (javierarce) [inline lib/assets/test/spec/cartodb3/editor/style/style-converter-fixtures.spec.js]: Ups, sorry, just pushed the fix for this.

File: lib/assets/test/spec/cartodb3/editor/style/style-converter-fixtures.spec.js (hunk 4 above)
Comment (xavijam) [inline lib/assets/test/spec/cartodb3/editor/style/style-converter-fixtures.spec.js]: No luck with this one?

**Pull request merged**